benches/cub/device/merge_sort/custom_less.cu

```diff
@@ -0,0 +1,97 @@
+#include <nvbench/nvbench.cuh>
+
+#include <curand.h>
+#include <curand_kernel.h>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+
+#include <cub/device/device_merge_sort.cuh>
+
+/*
+ * Arithmetic types with the custom comparator lead to merge sort usage.
+ */
+
+class less_comparator
+{
+public:
+  template <typename T>
+  __device__ bool operator()(T i, T j) noexcept
+  {
+    return i < j;
+  }
+};
+
+__global__ void rng_setup_kernel(std::size_t elements, curandState *states)
+{
+  unsigned int i = threadIdx.x + blockDim.x * blockIdx.x;
+
+  if (i < elements)
+    curand_init(1234, i, 0, states + i);
+}
+
+template <typename T>
+__global__ void rng_generate_kernel(const unsigned int n,
+                                    curandState *states,
+                                    T *result)
+{
+
+  const unsigned int i = threadIdx.x + blockDim.x * blockIdx.x;
+
+  if (i < n)
+  {
+    result[i] = static_cast<T>(ceilf(curand_uniform(states + i) * n));
+  }
+}
+
+template <typename T>
+void custom_less(nvbench::state &state, nvbench::type_list<T>)
+{
+  const auto elements = static_cast<std::size_t>(state.get_int64("Elements"));
+
+  thrust::device_vector<T> data(elements);
+  thrust::device_vector<curandState> rng_states(elements);
+
+  size_t temp_size = 0;
+  cub::DeviceMergeSort::SortKeys(
+    nullptr,
+    temp_size,
+    thrust::raw_pointer_cast(data.data()),
+    elements,
+    less_comparator());
+
+  thrust::device_vector<char> tmp(temp_size);
+
+  const unsigned int rng_threads_in_block = 256;
+  const unsigned int rng_blocks_in_grid =
+    (elements + rng_threads_in_block - 1) / rng_threads_in_block;
+
+  rng_setup_kernel<<<rng_blocks_in_grid, rng_threads_in_block>>>(
+    elements,
+    thrust::raw_pointer_cast(rng_states.data()));
+  cudaDeviceSynchronize();
+
+  state.exec([&](nvbench::launch &launch) {
+    rng_generate_kernel<<<rng_blocks_in_grid, rng_threads_in_block>>> (
```

Comment:
You'll want an explicit timer here. This currently includes the `rng_generate_kernel` execution in the time measurement.

See `nvbench/examples/exec_tag_timer.cu` for an example.

Reply:
Won't it disable batched measurement? My idea was to trade a constant workload with noise reduction.

Reply:
I'm thinking about introducing an overload of the merge sort so that the input array won't be modified. In this case, we won't affect the measured code segment with extra work.